src/app/components/shared/datetime/abstract-datetime.component.ts

```diff
@@ -0,0 +1,29 @@
+import { OnChanges, Component } from "@angular/core";
+import { NgbTooltipModule } from "@ng-bootstrap/ng-bootstrap";
+
+// use use a standalone component here so that it implements all its
+@Component({
+  selector: "baw-abstract-datetime",
+  templateUrl: "./abstract-datetime.component.html",
+  standalone: true,
+  imports: [NgbTooltipModule],
+})
+export abstract class AbstractDatetimeComponent implements OnChanges {
```

Comment:
comment truncated?

why is this a component at all? It's never used anywhere? ~~Just remove the decorator?~~

So it does need a decorator, but the extra information here is essentially unused? So don't repeat it.

Reply:
Sadly, there is no way to remove the templateUrl: "" as all angular components must have a template (there is no such thing as an abstract component without a template).

Therefore, the best I can do is
![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/fc1d2dc7-6fd1-4283-b6cf-9acbdd386e81)

Without hacking something together using abstract directives